executor/containerdexecutor/executor.go

```diff
@@ -193,61 +194,23 @@ func (w *containerdExecutor) Run(ctx context.Context, id string, root cache.Moun
 
 	task, err := container.NewTask(ctx, cio.NewCreator(cioOpts...), containerd.WithRootFS(rootMounts))
 	if err != nil {
-		return err
+		return status, err
 	}
+
 	defer func() {
 		if _, err1 := task.Delete(context.TODO()); err == nil && err1 != nil {
 			err = errors.Wrapf(err1, "failed to delete task %s", id)
 		}
 	}()
 
-	if err := task.Start(ctx); err != nil {
-		return err
-	}
-
-	if started != nil {
+	return w.runProcess(ctx, task, process.Resize, func() {
 		startedOnce.Do(func() {
 			close(started)
 		})
-	}
-	statusCh, err := task.Wait(context.Background())
-	if err != nil {
-		return err
-	}
-
-	var cancel func()
```

Comment:
all this wait & error handling code I moved to a common `runProcess` function to share between `Run` and `Exec`.  (I failed to realize the previous Exec usage was not synchronous and just returned immediately after starting, whoops)